harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -657,6 +657,61 @@ def test_amp(self, trial_class) -> None:
         )
         controller.run()
 
+    @pytest.mark.skipif(not HAVE_APEX, reason="Apex not available")
+    @pytest.mark.skipif(not torch.cuda.is_available(), reason="no gpu available")
+    @pytest.mark.gpu
+    def test_apex_checkpointing(self, tmp_path: pathlib.Path) -> None:
+        # There will be num_checkpoints-1 assertions made at the end of this test
+        num_checkpoints = 5
```

Comment:
question: what does num_checkpoints=5 buy us that num_checkpoints=2 wouldn't buy us?

Reply:
Cool, so let's pick to the minimum num_checkpoints to achieve full coverage.  Otherwise we're just wasting time in tests.

Reply:
This code is gone.